opal/mca/threads/argobots/threads_argobots_event.c

```diff
@@ -0,0 +1,143 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2019      Sandia National Laboratories.  All rights reserved.
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "opal/mca/threads/threads.h"
+#include "opal/mca/threads/argobots/threads_argobots.h"
+#include "opal/mca/event/libevent2022/libevent/include/event2/thread.h"
```

Comment:
Thank you for your comment. These `#include`s are needed to implement ULT-aware libevent code in `opal/mca/threads`. We admit that this is a temporary solution. Eventually this libevent wrapper should be removed as the official libevent (or maybe its fork) starts to support Argobots/Qthreads libraries.

If it is okay, I will keep this code but clarify this temporary fix by adding comments in `opal/mca/threads/argobots/threads_argobots_event.c` and `opal/mca/threads/README.md`. Otherwise, I will drop this libevent hack from this PR (and instead develop and use a Argobots/Qthreads-aware libevent library).

Reply:
No, I'm sorry -- this isn't ok.  We explicitly disallow including the header files of another component (except for `common` components -- those are specifically allowed to be used by other components).  It breaks all kinds of assumptions.

Yes, libevent2022 is almost certainly guaranteed to be there and valid, but if/when we upgrade libevent2022 is upgraded someday, this code will break, and it will be painful to find/fix.  More generally: "temporary" things like this have a way of never getting fixed over time.

I know this may seem nit-picky, but this is one of our golden rules that really should not be broken.  Sorry!

Is it a ton of work to make a libevent-aware common library between Argobots/Qthreads?

Reply:
Thank you. I understand it.I will make a patch to remove the libevent hack from this PR.